tests/ui/impl-trait/in-trait/issue-102140.stderr

```diff
@@ -6,7 +6,11 @@ LL |         MyTrait::foo(&self)
    |         |
    |         required by a bound introduced by this call
    |
-   = help: the trait `MyTrait` is implemented for `Outer`
+help: consider removing the leading `&`-reference
+   |
+LL -         MyTrait::foo(&self)
+LL +         MyTrait::foo(self)
```

Comment:
Not exactly sure why this diagnostic has come back -- it's kinda sketchy, but also it's kinda implemented sketchily. I kinda don't want to bother, but I guess could be convinced otherwise.